templates/company-board/contracts/CompanyBoardTemplate.sol

```diff
@@ -38,97 +38,109 @@ contract CompanyBoardTemplate is BaseTemplate {
         _ensureMiniMeFactoryIsValid(_miniMeFactory);
     }
 
-    function prepareInstance(string _shareTokenName, string _shareTokenSymbol) public {
+    function prepareInstance(string _shareTokenName, string _shareTokenSymbol) external {
         (Kernel dao,) = _createDAO();
         MiniMeToken boardToken = _createToken(BOARD_TOKEN_NAME, BOARD_TOKEN_SYMBOL, BOARD_TOKEN_DECIMALS);
         MiniMeToken shareToken = _createToken(_shareTokenName, _shareTokenSymbol, SHARE_TOKEN_DECIMALS);
         _storeCache(dao, boardToken, shareToken, msg.sender);
     }
 
     function setupInstance(
-        string _id, 
-        address[] _boardMembers, 
-        address[] _shareHolders, 
+        string _id,
+        address[] _boardMembers,
+        address[] _shareHolders,
         uint256[] _shareStakes,
-        uint64[] _boardVoteSettings, /* [voteDuration, supportRequired, minAcceptanceQuorum] */
-        uint64[] _shareVoteSettings, /* idem */
+        uint64[3] _boardVotingSettings, /* [voteDuration, supportRequired, minAcceptanceQuorum] */
+        uint64[3] _shareVotingSettings, /* [voteDuration, supportRequired, minAcceptanceQuorum] */
```

Comment:
I'm curious what solidity does with this extra information about the array's length. Does it enforce that it's only of length up to 3?

Reply:
You will get an `Out of bounds array access` compilation error if you try sth like `array[4]` somewhere... additionally, not sure if truffle or web3 do some kind of validation for calls